packages/expo-updates/android/src/main/java/expo/modules/updates/errorrecovery/ErrorRecovery.kt

```diff
@@ -0,0 +1,104 @@
+package expo.modules.updates.errorrecovery
+
+import android.os.Handler
+import android.os.HandlerThread
+import android.util.Log
+import com.facebook.react.ReactInstanceManager
+import com.facebook.react.bridge.DefaultNativeModuleCallExceptionHandler
+import com.facebook.react.bridge.ReactMarker
+import com.facebook.react.bridge.ReactMarkerConstants
+import com.facebook.react.devsupport.DisabledDevSupportManager
+import java.lang.ref.WeakReference
+import kotlin.Exception
+
+class ErrorRecovery {
+  internal val handlerThread = HandlerThread("expo-updates-error-recovery")
+  internal var handler: Handler? = null
```

Comment:
Make this nonnullable and mark as lateinit https://kotlinlang.org/docs/properties.html#late-initialized-properties-and-variables